logging/exp/levels.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exp
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slog"
+)
+
+// Level is a local copy of a slog.Level.
+type Level = slog.Level
+
+const (
```

Comment:
This particular file is pretty significantly over-engineered to ensure we only allocate the level names and slog values once.